nexus/src/app/background/tasks/instance_updater.rs

```diff
@@ -139,33 +129,32 @@ impl InstanceUpdater {
         while let Some(saga_result) = sagas.join_next().await {
             match saga_result {
                 Err(err) => {
-                    debug_assert!(
-                        false,
-                        "since nexus is compiled with `panic=\"abort\"`, and \
-                         we never cancel the tasks on the `JoinSet`, a \
-                         `JoinError` should never be observed!",
-                    );
-                    stats.sagas_failed += 1;
-                    last_err = Err(err.into());
+                    const MSG: &'static str = "since nexus is compiled with \
+                     `panic=\"abort\"`, and we never cancel the tasks on the \
+                     `JoinSet`, a `JoinError` should never be observed!";
+                    debug_assert!(false, "{MSG}");
+                    error!(opctx.log, "{MSG}"; "error" => ?err);
```

Comment:
not adding to `saga_errors` here?

Reply:
I thought about it, but this really should never happen. Also, because `saga_errors` is a `Vec` of `(Uuid, String)`s rather than just strings, we couldn't put this error in it, because we don't know the instance ID that the panicked saga task would correspond to.

We could change `saga_errors` to a `Vec<String>` and just format the UUID in it when there is one, but having it be structured data felt a little nicer...if you think it's important to include a `JoinError` here in the OMDB output, though, I would be happy to change it!

Reply:
I agree about the "never should happen" part, though it would be a shame to miss it because it's buried in logs instead of visible in the omdb output.

I'm ok leaving it out, but if you did want to include it, you could do:

`saga_errors: Vec<ThisParticularSagasErrors>`

and

```
enum ThisParticularSagasErrors {
  JoinError,
  UpdateError { instance: Uuid, error: String },
}
```

maybe?

Reply:
hmm, yeah, we could do that --- another option is to just rename `query_errors` to "errors" or something and just stick it in there instead.

Reply:
Okay, I went and fixed it in 245f16775d6b2964fcd493ef5b17c43d6e06b8a0